### ISSUE TITLE
.NET - supporting .nuspec and Directory.Build.props?

### ISSUE THREAD

**Issue by skolima** [dependabot/dependabot-core#1182]
Related to #541 , there's two more files related to NuGet updates: `.nuspec` and `Directory.Build.props`. Would you consider updating those as well in the future?

Disclosure: I'm one of the devs at https://github.com/NuKeeperDotNet/NuKeeper , and I think it's **awesome** dependabot got acquired by GitHub and it would be amazing to have feature parity 😄 Unfortunately, my Ruby skills are nonexistent, I'd be more than happy to answer any obscure NuGet/.NET/MSBuild questions.

So: `.nuspec` : this is the file describing how a NuGet package is published from a project. The "modern" .NET projects using `<PackageReference/>` format will generally have `.nuspec` generated automatically and there will be nothing to update. The older projects might have a version range specified there (not that common, AFAIK) or a fixed version, in line with what is in `packages.config`. In that last case, this file would need to be updated to stay in sync with `packages.config`.
[MSDN `.nuspec` file reference](https://docs.microsoft.com/en-us/nuget/reference/nuspec) 
[NuKeeper `.nuspec` update code](https://github.com/NuKeeperDotNet/NuKeeper/blob/bc357aaeacbaaad40c422091fcd7d26ae402db7b/NuKeeper.Update/Process/UpdateNuspecCommand.cs)

`Directory.Build.props` on the other hand is relevant only to the "modern" project format: it's a file automatically merged with every project in its folder and subfolders. Quite useful for specifying things like compilation settings, copyrights, versions - and shared references, like analyzers. This file can be updated (and has the same syntax) as a `.csproj`.
[MSDN introduction (rather barebones)](https://docs.microsoft.com/en-us/visualstudio/msbuild/customize-your-build?view=vs-2019#directorybuildprops-and-directorybuildtargets)
[Sample use with shared build analyzer settings](https://github.com/NuKeeperDotNet/NuKeeper/blob/master/Directory.Build.props)
[NuKeeper `Directory.Build.props` update code](https://github.com/NuKeeperDotNet/NuKeeper/blob/bc357aaeacbaaad40c422091fcd7d26ae402db7b/NuKeeper.Update/Process/UpdateDirectoryBuildTargetsCommand.cs)

**Comment by david-driscoll** [comment]
Any progress on this? 😄

**Comment by greysteil** [comment]
We already fetch `Directory.Build.props` files and update them, but not `.nuspec` ones (yet).

**Comment by greysteil** [comment]
And more generally, thanks for opening this @skolima, and for all your work on NuKeeper. I'm super keen to get to feature parity, but we're still a tiny team on Dependabot (we don't have much additional resource from GitHub yet and have a *lot* of additional work integrating directly into GitHub and dealing with a 1000x scale increase). We'll definitely get to this and PRs are very appreciated (although I understand not easy since Dependabot Core is in Ruby).

**Comment by skolima** [comment]
Update regarding the `Directory.Build.props`: there's a (minor) bug where currently Dependabot does not detect if that dependency has been updated externally and the PR is no longer needed, like it does for other updates - the PR is not auto-flagged and closed. Mentioning it with `@dependabot rebase` will correctly detect and close.

Repro repository: https://github.com/skolima/dependabot-props-problem/pull/3

**Comment by greysteil** [comment]
Good catch! Fixed by https://github.com/dependabot/dependabot-core/commit/55aab505de979d4188a7f3a9c18195190f356004. Thanks for the heads up.

**Comment by SeanFeldman** [comment]
@greysteil, is this issue still open as there's more to be done or can it be closed?

**Comment by greysteil** [comment]
I _think_ there's more to be done - the fix commit I mentioned above was for a small issue @skolima pointed out, not the full details of the issue body.

I don't work on Dependabot anymore (these days I'm a product manager for code security at GitHub, which is pretty much all our security stuff _except_ dependency related things), but you're in safe hands with `@feelepxyz` and the rest of the team.

**Comment by SeanFeldman** [comment]
Thank you, @greysteil.

@feelepxyz, could you please confirm that Depedabot respects `Directory.Build.props`? And if not, clarify what's missing? Thank you.

**Comment by feelepxyz** [comment]
@SeanFeldman dependabot should fetch and update `Directory.Build.props` files, the issue is also asking to update `.nuspec` which isn't currently supported.

**Comment by SeanFeldman** [comment]
Right. Perhaps considering an update to the issue description to have a list with `Directory.Build.props`checked off and `. nuspec` not would make it easier to see at a glance what’s the status? Or, closing this issue with a link to a specific one for `.nuspec`only?
Regardless, thank you @feelepxyz for the clarification.

**Comment by Jaxelr** [comment]
I'm not sure if i should open a new issue for this, but feel free to point me in the correct direction, i posted it here since this issue seems to be connected to the usability of `Directory.build.props` 

I have this repository that has the following structure:

![image](https://user-images.githubusercontent.com/5913008/135522921-e9183c18-68fc-425c-9da7-77dd2adc7b67.png)

And inside the OpenCar folder a .csproj:

![image](https://user-images.githubusercontent.com/5913008/135522968-852e38be-29e9-41eb-9d92-a27dc1a540c9.png)

ive configured the following dependabot.yml file with the structure:

```yaml
version: 2
updates:
  - package-ecosystem: "nuget"
    directory: "/OpenCar" #OpenCar.csproj
    schedule:
      interval: "weekly"
      day: "monday"
    commit-message:
      prefix: "deps"
    open-pull-requests-limit: 5
    labels:
      - "dependencies"
  - package-ecosystem: "nuget"
    directory: "." #Directory.build.props
    schedule:
      interval: "weekly"
      day: "saturday"
    commit-message:
      prefix: "deps"
    open-pull-requests-limit: 5
    labels:
      - "dependencies"
```

For the csproj it works as intended, but for reading the `Directory.builds.props` i get the following error message

![image](https://user-images.githubusercontent.com/5913008/135523295-653d8550-844e-4bb2-beb5-1d1d33e6ecc3.png)

Which would exclude the .props file from being used as a proper entry file.

**Comment by pellared** [comment]
I had the same issue as @Jaxelr . Dependabot only bumps `Directory.Build.props` located in the same directory where `.*proj` is present.

I came up with following workaround but I find it inconvinent: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/698

Can you also bump packages in `Directory.Build.props` that are located higher in directory structures?

**Comment by Saibamen** [comment]
.
